manifests/entry.pp

```diff
@@ -1,8 +1,8 @@
 define backupninja::entry (
-  $type,
-  $options,
-  $ensure = $backupninja::ensure,
-  $weight = '',
+  Enum['duplicity', 'ldap', 'mysql', 'pgsql', 'sh'] $type,
+  Hash $options,
+  Enum['present', 'absent'] $ensure = $backupninja::ensure,
+  String $weight = '',
```

Comment:
Understood, however `$weight` has a similar condition as `$when` in `entry.pp`:
```puppet
weight             => $weight ? {
          ''      => undef,
          default => $weight,
        },
```

To circumvent this problem I have changed the condition. We want to explicitly be an Integer, so I have changed it's default value to `-1`, since the weight of a backupninja task shouldn't be a negative number and changed the condition accordingly in a new commit.

If you agree with this change we can consider this PR solved.

Reply:
Probably here  we should try using Optional and passing the value untouched

Reply:
Or use String in the other types (I didn't look into details just point the inconsistence)

Reply:
The only uses I can see of weight are as String (prefixing files)

Reply:
Understood, all instances of `$weight` parameter have been changed to String. I can confirm that the only use the parameter has is as the file prefix so it does make sense to declare it as a String.

Reply:
In the end I reverted the weight into an Integer type, and changed the conditions in `entry.pp` accordingly:
```puppet
weight             => $weight ? {
          ''      => undef,
          default => $weight,
        },
```

MR core: https://gitlab.sbit.io/sbitio/core/-/merge_requests/440

Reply:
IMHO, if typed as integer, it must be optional on entry.pp